carbon-design-system/carbon
# refactor(react): update ModalHeader to functional component

Ref #9712 

Refactors the `ModalHeader` component in `ComposedModal` from a class to a functional component. 


#### Changelog

#### New

- Add functional component for `ModalHeader` under ComposedModal/next/ComposedModal
- Added new tests for next components

#### Testing / Reviewing

- Run yarn start:v11 from packages/react

- Ensure that `ComposedModal` story is working.
